### ISSUE TITLE
expect_fun_call should allow static string variables

### ISSUE THREAD

**Issue by sunjay** [rust-lang-nursery/rust-clippy#3017]
I have some code [here](https://github.com/sunjay/turtle/blob/9240f8890d1032a0033ec5c5338a10ffa942dc21/src/messenger.rs#L61) which calls `expect` but passes in a string literal provided by a function argument. Clippy complains and says:

```
warning: use of `expect` followed by a function call
  --> src/messenger.rs:61:35
   |
61 |         .map(|_| writeln!(writer).expect(unable_to_write_newline))
   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(|_| panic!(unable_to_write_newline))`
   |
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#expect_fun_call
```

This isn't a function call though. In fact, it won't even have the performance penalty that this lint is trying to avoid.

Here's a smaller reproduction of the same error: ([Playground Link](http://play.rust-lang.org/?gist=ec27eac1f316cbd7f89f6e4ea2d78715&version=stable&mode=debug&edition=2015))

```rust
fn foo(res: Result<(), ()>, message: &'static str) {
    res.expect(message);
}

fn main() {
    foo(Ok(()), "foo");
}
```

This produces:

```
warning: use of `expect` followed by a function call
 --> src/main.rs:2:9
  |
2 |     res.expect(message);
  |         ^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(|_| panic!(message))`
  |
  = note: #[warn(expect_fun_call)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#expect_fun_call
```

I think the best thing to do is to not produce this warning if the code is like this. Doing it for a function call or a call to `format!` makes sense, but it doesn't make sense for variables that are static strings.

**Comment by mikerite** [comment]
I believe this is the same error as #2979 which was fixed by #2991. I guess the next version of Clippy won't have this bug.

**Comment by sunjay** [comment]
@mikerite Yup! Thanks!
